src/validation/fields/helper/date.validation.helper.ts

```diff
@@ -43,6 +44,41 @@ export const conditionalDateValidations = (dateContextWithCondition: dateContext
   ];
 };
 
+export const conditionalHistoricalBODateValidations = (trustDateContext: dateContextWithCondition) => [
+  setValidateCeasedDateFlag(trustDateContext),
+  ...conditionalDateValidations(trustDateContext),
+];
+
+// export const setValidateCeasedDateFlag = (trustDateContext: dateContextWithCondition) => [
+//   body()
+//     .custom((value, { req }) => {
+//       const isUnableToProvideAllTrustInfo: boolean = isUnableToObtainAllTrustInfo(req);
+//       let validateCeasedDate = true;
+//       if ((req.body[trustDateContext.dayInput.name] === "" && req.body[trustDateContext.monthInput.name] === "" && req.body[trustDateContext.yearInput.name] === "") && isUnableToProvideAllTrustInfo) {
+//         validateCeasedDate = false;
+//       }
+//       req.body['validateCeasedDate'] = validateCeasedDate.toString();
+//       return true;
+//     }),
+// ];
+// export const getApplicationData = (session: Session | undefined): ApplicationData => {
+//   return session?.getExtraData(APPLICATION_DATA_KEY) || {} as ApplicationData;
+// };
+export const setValidateCeasedDateFlag = (trustDateContext: dateContextWithCondition) => {
+  console.log("Hello!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!");
+  return (req, resp, next: NextFunction) => {
```

Comment:
you should also add the types for req and res into the parameters like you have for NextFunction

Reply:
Added types